packages/token/hardhat.config.ts

```diff
@@ -0,0 +1,57 @@
+import { HttpNetworkUserConfig } from "hardhat/types";
+import { HardhatUserConfig } from "hardhat/config";
+import "@nomicfoundation/hardhat-toolbox";
+import "hardhat-deploy";
+
+// read MNEMONIC from env variable
+let mnemonic = process.env.MNEMONIC;
+
+const infuraNetwork = (
+    network: string,
+    chainId?: number,
+    gas?: number
+): HttpNetworkUserConfig => {
+    return {
+        url: `https://${network}.infura.io/v3/${process.env.PROJECT_ID}`,
+        chainId,
+        gas,
+        accounts: mnemonic ? { mnemonic } : undefined,
+    };
+};
+
+const config: HardhatUserConfig = {
+    solidity: "0.8.18",
```

Comment:
[Solidity 0.8.20](https://blog.soliditylang.org/2023/05/10/solidity-0.8.20-release-announcement/) is out, and it has an interesting gas optimization (they added a `PUSH0` opcode).
We could try using it and look for any improvement in gas costs.